aten/src/ATen/cpu/vec256/vec256_base.h

```diff
@@ -307,6 +308,15 @@ template <class T> Vec256<T> inline operator/(const Vec256<T> &a, const Vec256<T
   return c;
 }
 
+template <class T> Vec256<T> inline operator&&(
+    const Vec256<T> &a, const Vec256<T> &b) {
+  Vec256<T> c = Vec256<T>();
+  for (int i = 0; i != Vec256<T>::size(); i++) {
+    c[i] = a[i] && b[i];
```

Comment:
I don't think it's a good idea to add this to Vec256:

1) The behavior differs from the other "boolean" type operations (such as comparisons). In those, the result for true is all ones (which is -1 for integers and a NAN for floats). Here the true result is part zeros part ones.

2) There's only a non vectorized version, which means that it's better to not use Vec256 here at all.

I realize it's difficult to avoid the Vec256 because binary_kernel_reduce_vec expects it. I see two possible options:

a) Add a `binary_kernel_reduce` that only operates on scalars. You can use `binary_kernel_reduce_vec` as a starting point

b) Add this only to and_kernel_impl instead of Vec256. Even if it's slower than it needs to be, at least it doesn't add an operator that has inconsistent behavior with the other operations.

Reply:
Thanks, will address.

Question:

Regarding the inconsistency, I think the code might still be correct if we change `&&` to return `-1` as other logical operations, as eventually it reduces to a single scalar, which will use the scalar version `&&` (haven't tried). The downside is that is it using an assumption on the internal implementation of `binary_kernel_reduce` which is not explicitly advertised.  Is this acceptable?